githubdelivery/src/main/java/io/dockstore/githubdelivery/GithubDeliveryHelper.java

```diff
@@ -0,0 +1,62 @@
+package io.dockstore.githubdelivery;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.gson.JsonSyntaxException;
+import io.dockstore.openapi.client.model.InstallationRepositoriesPayload;
+import io.dockstore.openapi.client.model.PushPayload;
+import io.dockstore.openapi.client.model.ReleasePayload;
+import static io.dockstore.utils.ExceptionHandler.exceptionMessage;
+import java.io.IOException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.services.s3.model.NoSuchKeyException;
+
+/**
+ * Helper for deserializing GitHub events in JSON to Java objects, as well as some logging
+ * methods.
+ */
+final class GithubDeliveryHelper {
+
+    private static final Logger LOG = LoggerFactory.getLogger(GithubDeliveryHelper.class);
+    
+    private static final ObjectMapper MAPPER = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
+    private GithubDeliveryHelper() {
+    }
+
+    static PushPayload getGitHubPushPayloadByKey(String eventType, String body, String key) throws IOException, NoSuchKeyException {
+        return getPayloadByKey(eventType, body, key, PushPayload.class);
+    }
+
+    static InstallationRepositoriesPayload getGitHubInstallationRepositoriesPayloadByKey(String eventType, String body, String key) throws IOException, NoSuchKeyException {
+        return getPayloadByKey(eventType, body, key, InstallationRepositoriesPayload.class);
+    }
+
+    static ReleasePayload getGitHubReleasePayloadByKey(String eventType, String body, String key) throws IOException, NoSuchKeyException {
+        return getPayloadByKey(eventType, body, key, ReleasePayload.class);
+    }
+
+    static <T> T getPayloadByKey(String eventType, String body, String key, Class<T> clazz) throws IOException,
+        NoSuchKeyException {
+        try {
+            T payload = MAPPER.readValue(body, clazz);
+            if (payload == null) {
+                logReadError(eventType, key);
+            }
+            return payload;
+        } catch (JsonSyntaxException e) {
+            exceptionReadError(e, eventType, key);
+        }
+        return null;
+    }
```

Comment:
Some thoughts, more for the sake of brain dumping than prescribing changes (most of this relates to constructs that existed before this PR):
1. The `return null` is because the compiler can't tell that `exceptionReadError` exits, correct?
2. Would be nice to swallow the `IOException`, we're not doing any IO here so it should never be thrown (right?) and it'd be great to not have to deal with it in the consuming code.
3. Can this method throw a `NoSuchKeyException`?
4. `exceptionReadError` exits, but do we want the utility to completely stop if an event's json is corrupt or non-standard?  Could log and keep going.  `exceptionReadError` delegates to `exceptionMessage`, which feels like its intended to be used in the outer `catch` of a utility, as a standard way to generate a message and exit.

Reply:
As background, this was mainly existing code, which I just refactored to avoid triplication -- there was code that deserialized into a PushPayload, almost the same code for InstallationRepositoriesPayload, and I was adding deserialization for  ReleasePayload.

So I made this base method, but I kept the existing behavior, as we don't have tests beyond what I just created here, for me to change other behavior with confidence. I had to put the method, and the methods below, to a new class, because GithubDeliveryClientS3 requires AWS configuration, which is tricky to configure for unit tests.

1. It was returning null before, so I kept that.
2. You're probably right, but I think I'll keep it, just because I don't understand the implications.
3. You're right, it doesn't throw one. Removed.
4. You're probably right, but I don't want to mess with the overall flow at this point.

My justification for not doing some of this stuff is that this is "just" an internal utility to redeliver s3 events that we hopefully won't need to use, is not user-facing code, and I'm just adding support for a new event type, and don't think it's worth the effort/time to refactor further at this point.